cmd/vcert/commands.go

```diff
@@ -756,6 +693,137 @@ func doCommandCredMgmt1(c *cli.Context) error {
+			fmt.Println("access_token_expires: ", tm)
+			fmt.Println("refresh_token: ", resp.Refresh_token)
+			fmt.Println("refresh_until: ", time.Unix(int64(resp.Refresh_until), 0).UTC().Format(time.RFC3339))
+		}
+	} else if cfg.Credentials.User != "" && cfg.Credentials.Password != "" {
+
+		auth := &endpoint.Authentication{
+			User:     cfg.Credentials.User,
+			Password: cfg.Credentials.Password,
+			Scope:    flags.scope,
+			ClientId: flags.clientId}
+
+		if flags.sshCred {
+			auth.Scope = "ssh:manage"
+		} else if flags.pmCred {
+			auth.Scope = "certificate:manage,revoke;configuration:manage"
+		}
+
+		resp, err := tppConnector.GetRefreshToken(auth)
+		if err != nil {
+			return err
+		}
+		if flags.credFormat == "json" {
+			if err := outputJSON(resp); err != nil {
+				return err
+			}
+		} else {
+			tm := time.Unix(int64(resp.Expires), 0).UTC().Format(time.RFC3339)
+			fmt.Println("access_token: ", resp.Access_token)
+			fmt.Println("access_token_expires: ", tm)
+			if resp.Refresh_token != "" {
+				fmt.Println("refresh_token: ", resp.Refresh_token)
+				fmt.Println("refresh_until: ", time.Unix(int64(resp.Refresh_until), 0).UTC().Format(time.RFC3339))
+			}
+		}
+	} else if clientP12 {
+		resp, err := tppConnector.GetRefreshToken(&endpoint.Authentication{
+			ClientPKCS12: clientP12,
+			Scope:        flags.scope,
+			ClientId:     flags.clientId})
+		if err != nil {
+			return err
+		}
+		if flags.credFormat == "json" {
+			if err := outputJSON(resp); err != nil {
+				return err
+			}
+		} else {
+			tm := time.Unix(int64(resp.Expires), 0).UTC().Format(time.RFC3339)
+			fmt.Println("access_token: ", resp.Access_token)
+			fmt.Println("access_token_expires: ", tm)
+			if resp.Refresh_token != "" {
+				fmt.Println("refresh_token: ", resp.Refresh_token)
+				fmt.Println("refresh_until: ", time.Unix(int64(resp.Refresh_until), 0).UTC().Format(time.RFC3339))
+			}
+		}
+	} else {
+		return fmt.Errorf("failed to determine credentials set")
+	}
+
+	return nil
+}
+
+func getVaaSCredentials(vaasConnector *cloud.Connector, cfg *vcert.Config) error {
+	//TODO: quick workaround to suppress logs when output is in JSON.
+	if flags.credFormat != "json" {
+		logf("Getting credentials...")
+	}
+
+	if cfg.Credentials.User != "" {
+
+		statusCode, userDetails, error := vaasConnector.CreateAPIUserAccount(cfg.Credentials.User, cfg.Credentials.Password)
+
+		if error != nil {
+			return fmt.Errorf("failed to create a User Account/rotate API Key in VaaS: %s", error)
+		}
+
+		apiKey := userDetails.APIKey
+
+		if flags.credFormat == "json" {
+			if err := outputJSON(apiKey); err != nil {
+				return err
+			}
+		} else {
+			var headerMessage string
+			if statusCode == http.StatusCreated {
+				headerMessage = "the user account was created successfully. To complete the registration please review your email account and follow the link."
+			} else { // it's assumed the statusCode is 202 which means the user exists and therefore the APIKey is rotated
```

Comment:
You should not assume the statusCode value and double check that it is indeed 202.

Reply:
The new function `parseUserDetailsResultFromPOST pkg/venafi/cloud/cloud.go line 384` which is the responsible to parse the response from VaaS only returns successfully if the statuscode is 201 or 202 so therefore it's redundant to do a validation to 202, but to avoid confusion on the understanding of the code logic maybe is a good idea to add the validation.

Reply:
Added the validation